scripts/ShipPlanner015.py

```diff
@@ -0,0 +1,682 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+# This is an addon for Blender, that provides some tools to calculate
+# ship stats, and exports it next to the blend file with the same name.
+
+import bpy
+import math
+import bpy_extras.io_utils
+import os
+
+bl_info = {
+	"name": "Pioneer Ship Planner",
+	"author": "Szilard Balint (nozmajner)",
+	"version": (0,1,5),
+	"blender": (2, 76, 0),
+	"location": "Properties -> Scene -> Ship Planner",
+	"description": "Calculates ship stats for Pioneer Space Simulator",
+	"warning": "",
+	"wiki_url": "N/A",
+	"tracker_url": "N/A",
+	"category": "Scene"}
+
+gee = 9.81
+def sum_mass(hull, fuel, cap, cargo_cap):	
+	result = [hull+fuel+cap, hull+fuel, hull+fuel+cargo_cap] 
+	return result
+
+def acc(mass, thrust):
+	acc_full = (thrust/mass[0]/gee) #mass*1000 volt
+	acc_empty = (thrust/mass[1]/gee)
+	
+	if acc_full <= 0.0001:
+		result_full = "<0.0001"
+	else:
+		result_full = acc_full
+	
+	if acc_empty <= 0.0001:
+		result_empty = "<0.0001"
+	else:
+		result_empty = acc_empty
+		
+	result = [result_full, result_empty]
+		
+	return result
+
+def deltav(ev, mass, fuel_cap):
+	dv = (float(ev))*(math.log(float(mass+fuel_cap)/mass))
+	if dv <=0.0001:
+		result = "<0.0001"
+	else:
+		result = dv
+	return result
+
+
+ #Operator example
+class VesselExport(bpy.types.Operator): # , bpy_extras.io_utils.ExportHelper .ExportHelper feldobja az export ablakot, ha megnyomom a gombot
+	"""Export shipdef.lua"""
+	bl_idname = "object.simple_operator"
+	bl_label = "Export shipdef"
+	filename_ext = ".lua"
+
+	@classmethod
+	def poll(cls, context):
+		return context.active_object is not None
+
+	def execute(self, context):
+		print("Krumpli") 
+		scene = bpy.data.scenes["Scene"]
+#		file = open(filepath, 'w', encoding="ansi", newline = "\n")
+		file = open(os.path.splitext(bpy.data.filepath)[0] + ".json", 'w', newline = "\n")
```

Comment:
The other comments have their "#" indented. I guess that's more sane, considering python's indention-pickiness.